common/src/main/java/org/apache/celeborn/common/network/server/TransportChannelHandler.java

```diff
@@ -152,8 +153,18 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
 
   @Override
   public void channelRead(ChannelHandlerContext ctx, Object request) throws Exception {
-    if (request instanceof RequestMessage && !(request instanceof Heartbeat)) {
-      requestHandler.handle((RequestMessage) request);
+    if (request instanceof RequestMessage) {
+      if (request instanceof Heartbeat) {
+        logger.debug("Received heartbeat from {}.", NettyUtils.getRemoteAddress(ctx.channel()));
+        if (!enableHeartbeat) {
+          // Which disables heartbeat should send heartbeat for processing heartbeat to avoid
+          // reading idleness of which enables heartbeat.
+          requestHandler.processHeartbeat();
```

Comment:
`processHeartbeat` always results in sending a `Heartbeat` as response (not `HeartbeatResponse`).
Are we sure this does not result in both sides constantly sending Heartbeat's to each other ?

Reply:
@mridulm, there is no case that both sides constantly sending Heartbeat's to each other. I could give the following situations:

1. worker disables heartbeat and client disables heartbeat: the heartbeat message does not send.
2. worker enables heartbeat and client disables heartbeat: client invokes `processHeartbeat` and `fireChannelRead`, but worker only invoke `fireChannelRead`.
3. worker disables heartbeat and client enables heartbeat: worker invokes `processHeartbeat` and `fireChannelRead`, but client only invoke `fireChannelRead`.
4. worker enables heartbeat and client enables heartbeat: worker and client only invoke `fireChannelRead`.

Therefore, the heartbeat is only used to keep both sides of conection not read idle and does not constantly send heartbeat. PTAL.

Reply:
I was trying to understand if case 4 had corner cases. Makes sense, thanks for clarifying !